Loop/Managers/DeviceDataManager.swift

```diff
@@ -409,7 +408,8 @@ final class DeviceDataManager {
             alertManager: alertManager,
             analyticsServicesManager: analyticsServicesManager,
             loggingServicesManager: loggingServicesManager,
```

Comment:
This allows the `DeviceDataManager` to handle actions for the `ServicesManager` -- via.a new `RemoteCommandHandler` protocol. When a plugin receives an action, it needs to get that action to the DeviceDataManager.

Reply:
DeviceDataManager already exposes methods for dosing. Does it need separate methods for "remote" dosing?  If so, why?

Reply:
I'm wondering how `ServicesManager` can communicate with `DeviceDataManager`, assuming `DeviceDataManager` is still where dosing/settings things will live. Right now there is not a reference I see going that direction. So this delegate provides a bridge for the `ServicesManagerManager` to talk to the `DeviceDataManager` to enact doses, change settings, etc.

Is it the delegate name of `RemoteActionDelegate` that I should change? Like `DeviceDataManagerDelegate`?

Reply:
`ServicesManager` will still need to communicate with `DeviceDataManager` for dosing. But it should only be for dosing. I don't think DeviceDataManager needs to know about anything "remote".

Reply:
Ok I'll look at keeping the delegate, as we still need a bridge between these classes, but I'll rename it -- maybe `ServicesManagerDelegate` (I had the delegation relationship backwards in my prior comment).

Then I'll remove as much of the remote things from DeviceDataManager into the `ServicesManager` or `RemoteDataServicesManager`.

Do you want these changes as part of this PR set or a future one? This change seems big enough that it may be easier to test/review separately from all this. Whatever you prefer.

Reply:
I documented the relationships between these managers to get the big picture of how this update will fit.  A few notes on the plans for this update:

1. The `RemoteActionDelegate`, implemented by `DeviceDataManager` in this PR, will be renamed to `ServicesManagerDelegate` (red line in diagram). That will be for dosing/settings updates. I'll rename the delegate methods to drop `handleRemote` from prefix.
2. Update my dosing/settings error messages in `DeviceDataManager` to remove any 'Remote' phraseology to be consistent with the rest of this rearchitecture. I plan to not change anything else in those methods, including remote trigger uploads and analytics uploads (will stay in `DeviceDataManager`)
3 Move the background task code, all in support of remote, from `DeviceDataManager` to `ServicesManager`.
4. Move `handleRemoteNotification` from `DeviceDataManager` to `ServicesManager`. 

![Screenshot 2023-06-22 at 5 50 45 AM](https://github.com/LoopKit/Loop/assets/79173438/a7b0cf9f-3394-4d87-a941-b29d6aea2473)

Reply:
I documented the relationships between these managers to get the big picture of how this update will fit.  A few notes on the plans for this update:

1. The `RemoteActionDelegate`, implemented by `DeviceDataManager` in this PR, will be renamed to `ServicesManagerDelegate` (red line in diagram). That will be for dosing/settings updates. I'll rename the delegate methods to drop `handleRemote` from prefix.
2. Update my dosing/settings error messages in `DeviceDataManager` to remove any 'Remote' phraseology to be consistent with the rest of this rearchitecture. I plan to not change anything else in those methods, including remote trigger uploads and analytics uploads (will stay in `DeviceDataManager`)
3. Move the background task code, all in support of remote, from `DeviceDataManager` to `ServicesManager`.
4. Move `handleRemoteNotification` from `DeviceDataManager` to `ServicesManager`. 
5. 
![Screenshot 2023-06-22 at 5 55 54 AM](https://github.com/LoopKit/Loop/assets/3207996/d0c91615-7080-427a-a342-64233ce5b5e1)

Reply:
@ps2 I've made the changes that I describe in the last comment and pushed to this branch. Let me know if this is what you are looking for.

Note there is a still the `RemoteActionDelegate` (not represented in this diagram) that lives between the `RemoteDataServices` and `ServicesManager`. Then there is the new delegate between ServicesManager and DeviceDataManager which has no references to "Remote" like you mentioned.

I have not adjusted the naming in `RemoteActionDelegate` per your comments in the LoopKit PR. I'll look at that separately. I think that is the last of your feedback thus far.